helm/mattermod/templates/deployment.yaml

```diff
@@ -0,0 +1,169 @@
+  selector:
+    matchLabels:
+      {{- include "mattermod.selectorLabels" . | nindent 6 }}
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "mattermod.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "mattermod.serviceAccountName" . }}
+      terminationGracePeriodSeconds: 30
+      containers:
+        - name: {{ .Chart.Name }}
+          command:
+            - mattermod
+            - -config=/app/config/config.json
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: {{ .Values.service.port }}
+              protocol: TCP
+          livenessProbe:
+            failureThreshold: 3
+            httpGet:
+              path: /healthz
+              port: http
+              scheme: HTTP
+            initialDelaySeconds: 30
+            periodSeconds: 10
+            successThreshold: 1
+            timeoutSeconds: 5
+          readinessProbe:
+            failureThreshold: 3
+            httpGet:
+              path: /healthz
+              port: http
+              scheme: HTTP
+          volumeMounts:
+            - mountPath: /app/config/config.json
+              name: config
+              readOnly: true
+              subPath: config-mattermod.json
+            - mountPath: /app/.ssh
+              name: {{ template "mattermod.fullname" . }}-gitconfig
+              readOnly: true
+            - mountPath: /app/repos/
+              name: data-repos
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+          terminationMessagePath: /dev/termination-log
+          terminationMessagePolicy: File
+      {{- with .Values.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      volumes:
+        - configMap:
+            defaultMode: 420
+            name: mattermod-config
+          name: config
+        - name: {{ template "mattermod.fullname" . }}-gitconfig
+          secret:
+            defaultMode: 256
+            secretName: {{ template "mattermod.fullname" . }}-gitconfig
+        - name: data-repos
+          persistentVolumeClaim:
+            claimName: {{ template "mattermod.fullname" . }}-data
+
+---
+# Jobserver
+{{- if .Values.jobserver.enabled }}
```

Comment:
nit: why not create it a separate file `jobserver.yaml` to avoid long yaml file?

Reply:
Thought of having a more concise approach, with not a lot of files. Having [deplοyments,services,service-monitors] under the same file. 

We can easily seperate those, using a file named like `jobserver_deployment.yaml `
Same for `jobserver_deployment.yaml` and `jobserver-service-monitor.yaml`. 

@spirosoik  If you agree, will push a relevant commit.

Reply:
Done with: https://github.com/mattermost/mattermost-mattermod/pull/344/commits/76c8355d7951cb7a658effaef5861499641489dc